builtin/docker/builder.go

```diff
@@ -341,11 +353,16 @@ func (b *Builder) buildWithDocker(
 		return err
 	}
 
+	if platform != "" && ver != types.BuilderBuildKit {
+		return status.Errorf(codes.InvalidArgument, "buildkit is required to use platform option")
+	}
+
 	resp, err := cli.ImageBuild(ctx, buildCtx, types.ImageBuildOptions{
 		Version:    ver,
 		Dockerfile: relDockerfile,
 		Tags:       []string{tag},
 		Remove:     true,
+		Platform:	platform,
```

Comment:
This might be an editor thing to configure, but the spacing here was off (tab versus space). You could try a `go fmt` on this file to fix this block and align all the options in the meantime.